### PR TITLE
Collect and expose compile-only dependencies through ApplicationModel

### PR DESCRIPTION
**Pull request by aloubyansky**
with DependencyFlag.COMPILE_ONLY flag set (Maven provided scope)

FYI @ia3andy

### CODE REVIEW

**Comment by ia3andy** [comment]
thanks @aloubyansky it's perfect! (Just the flag combination issue to fix)

File: independent-projects/bootstrap/app-model/src/main/java/io/quarkus/bootstrap/model/ApplicationModel.java [hunk 1]
```diff
@@ -40,6 +40,14 @@ public interface ApplicationModel {
      */
     Iterable<ResolvedDependency> getDependencies(int flags);
 
+    /**
+     * Returns application dependencies that have any of the flags passed in as arguments set.
+     *
+     * @param flags dependency flags to match
+     * @return application dependencies that matched the flags
+     */
+    Iterable<ResolvedDependency> getDependenciesWithAnyFlag(int... flags);
```
Comment (aloubyansky) [inline independent-projects/bootstrap/app-model/src/main/java/io/quarkus/bootstrap/model/ApplicationModel.java:49]: @ia3andy this is the method you would call passing in `DependencyFlags.RUNTIME_CP, DependencyFlags.COMPILE_ONLY`.

**Comment by aloubyansky** [comment]
@ia3andy it's ready for the final review, afaict. We should merge it asap if we want it in 3.7/3.8.
FYI @gsmet

**Comment by ia3andy** [comment]
Awesome, I will review it today!

**Review by ia3andy — APPROVED** [review]
LGTM, tested with Maven and Gradle Web Bundler projects.

Thanks @aloubyansky

**Pull request merged**